src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -623,14 +729,49 @@ public bool HasSkipLocalsInitAttribute
         internal bool IsAutoPropertyWithGetAccessor
             => IsAutoProperty && _getMethod is object;
 
-        protected bool IsAutoProperty
-            => (_propertyFlags & Flags.IsAutoProperty) != 0;
+        internal bool IsAutoProperty
+        {
+            get
+            {
+                return (_propertyFlags & Flags.IsAutoProperty) != 0;
+            }
+        }
 
+#nullable enable
         /// <summary>
         /// Backing field for automatically implemented property, or
         /// for a property with an initializer.
         /// </summary>
-        internal SynthesizedBackingFieldSymbol BackingField { get; }
+        internal SynthesizedBackingFieldSymbol? BackingField
+        {
+            get
+            {
+                if (_lazyBackingFieldSymbol is SynthesizedBackingFieldSymbol { IsEarlyConstructed: true } backingField)
+                {
+                    return backingField;
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Backing field for a semi automatically implemented property.
+        /// </summary>
+        internal SynthesizedBackingFieldSymbol? FieldKeywordBackingField
```

Comment:
>FieldKeywordBackingField [](http://example.com/codeflow?start=48&length=24)

Only ```backingField.IsCreatedForFieldKeyword``` gives the answer whether a backing field is a ```field``` keyword backing field. Why does the implementation check something else?

Reply:
That's exactly what I was asking in https://github.com/dotnet/roslyn/pull/57076#issuecomment-1001013602. Will adjust this.

Reply:
> That's exactly what I was asking in [https://github.com/dotnet/roslyn/pull/57076#issuecomment-1001013602](https://github.com/dotnet/roslyn/pull/57076#issuecomment-1001013602).

I think there were clear instructions in [https://github.com/dotnet/roslyn/pull/57076#discussion_r775059877](https://github.com/dotnet/roslyn/pull/57076#discussion_r775059877).

Reply:
@AlekseyTs Yes it's absolutely very clear. I had missed the comment, sorry for that.

Reply:
@AlekseyTs This should be fixed now.